File: 2009/J092.cpp
#include <cstdio>
#include <cstring>
char s[20][21];
int v[20];
int main() {
  int n;
  scanf("%d", &n);
  for (int i = 0; i < n; i++) {
    scanf("%s", s[i]);
    v[i] = 0;
    for (int j = 0; j < strlen(s[i]); j++) {
      v[i] |= 1 << (s[i][j] - 'A');
    }
  }
  int mini, min = 21;
  for (int i = 0; i < 1 << n; i++) {
    int a = 0, count = 0;
    for (int j = 0; j < n; j++) {
      if (i & 1 << j) {
        a |= v[j];
        count++;
      }
    }
    if (a == (1 << 26) - 1 && count < min) {
      min = count;
      mini = i;
    }
  }
  if (min == 21) {
    printf("Impossible\n");
    return 0;
  }
  printf("%d\n", min);
  for (int j = 0; j < n; j++) {
    if (mini & 1 << j) {
      printf("%s\n", s[j]);
    }
  }
  return 0;
}

File: 2009/J092.pas
var
  n, i, j, a, count, mini, min: longint;
  s: array[0..20] of string;
  v: array[0..20] of longint;
begin
  readln(n);
  for i := 0 to n - 1 do
  begin
    readln(s[i]);
    v[i] := 0;
    for j := 1 to length(s[i]) do
      v[i] := v[i] or (1 shl (ord(s[i][j]) - ord('A')));
  end;
  min := 21;
  for i := 0 to 1 shl n do
  begin
    a := 0;
    count := 0;
    for j := 0 to n - 1 do
      if ((i and (1 shl j)) > 0) then
      begin
        a := a or v[j];
        inc(count);
      end;
    if ((a = (1 shl 26) - 1) and (count < min)) then
    begin
      min := count;
      mini := i;
    end;
  end;
  if (min = 21) then
    writeln('Impossible')
  else
  begin
    writeln(min);
    for j := 0 to n - 1 do
      if ((mini and (1 shl j)) > 0) then
        writeln(s[j]);
  end;
end.